File: 0530-Minimum-Absolute-Difference-in-BST/0530.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 * int val;
 * TreeNode left;
 * TreeNode right;
 * TreeNode() {}
 * TreeNode(int val) { this.val = val; }
 * TreeNode(int val, TreeNode left, TreeNode right) {
 * this.val = val;
 * this.left = left;
 * this.right = right;
 * }
 * }
 */
class Solution {
    private int min = Integer.MAX_VALUE;
    private Integer prev = null;

    public int getMinimumDifference(TreeNode root) {
        inorder(root);
        return min;
    }

    public void inorder(TreeNode node) {
        if (node == null)
            return;

        inorder(node.left);

        if (prev != null) {
            min = Math.min(node.val - prev, min);
        }

        prev = node.val;

        inorder(node.right);
    }
}

File: 0530-Minimum-Absolute-Difference-in-BST/0530.py
class Solution:
    def getMinimumDifference(self, root: Optional[TreeNode]) -> int: # type: ignore
        self.prev = None # To track the previous node's value
        self.min = 10 ** 5

        # In-order traversal: left -> current -> right
        def inorder(node):
            if not node: return

            inorder(node.left)

            # Calculate difference with the previous node
            if self.prev is not None:
                self.min = min(self.min, node.val - self.prev)
            
            self.prev = node.val

            inorder(node.right)
        
        inorder(root)

        return self.min